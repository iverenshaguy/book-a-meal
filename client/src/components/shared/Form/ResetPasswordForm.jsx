import React, { Fragment } from 'react';
import { RenderInput } from '../FormComponents';
import { formPropTypes } from '../../../helpers/proptypes';

/**
 * @exports
 * @function ResetPasswordForm
 * @param {object} props
 * @returns {JSX} ResetPasswordForm
 */
const ResetPasswordForm = ({ state, handlers }) => (
  <Fragment>
    <RenderInput
      type="password"
      name="password"
      label="Password"
      id="password"
      required
      value={state.values.password}
      placeholder=""
      handleChange={handlers.handleChange}
      handleBlur={handlers.handleBlur}
      handleFocus={handlers.handleFocus}
      meta={{
        touched: state.touched.password,
        error: state.error.password,
      }}
    />

    <RenderInput
      type="password"
      name="passwordConfirm"
      label="Confirm Your Password"
      id="passwordConfirm"
      required
      value={state.values.passwordConfirm}
      placeholder=""
      handleChange={handlers.handleChange}
      handleBlur={handlers.handleBlur}
      handleFocus={handlers.handleFocus}
      meta={{
        touched: state.touched.passwordConfirm,
        error: state.error.passwordConfirm
      }}
    />
  </Fragment>
);

ResetPasswordForm.propTypes = {
  ...formPropTypes('resetPassword')
};

export default ResetPasswordForm;
