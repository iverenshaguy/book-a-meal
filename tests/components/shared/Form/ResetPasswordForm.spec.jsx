import React from 'react';
import ResetPasswordForm from '../../../../src/components/shared/Form/ResetPasswordForm';
import { formComponentSetup } from '../../../setup/formSetup';

const token = 'fdghjklfghjklfghjklfdghjkl';

describe('ResetPasswordForm', () => {
  afterAll(() => {
    jest.clearAllMocks();
  });

  it('should render the Reset Password Form correctly', () => {
    const { handlers, state } = formComponentSetup('resetPassword');
    const wrapper = shallow(<ResetPasswordForm type="resetPassword" token={token} state={state} handlers={handlers} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
