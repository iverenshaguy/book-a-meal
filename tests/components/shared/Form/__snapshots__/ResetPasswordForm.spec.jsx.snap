// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPasswordForm should render the Reset Password Form correctly 1`] = `
<Fragment>
  <RenderInput
    handleBlur={[MockFunction]}
    handleChange={[MockFunction]}
    handleFocus={[MockFunction]}
    id="password"
    label="Password"
    meta={
      Object {
        "error": null,
        "touched": false,
      }
    }
    name="password"
    placeholder=""
    required={true}
    type="password"
    value=""
  />
  <RenderInput
    handleBlur={[MockFunction]}
    handleChange={[MockFunction]}
    handleFocus={[MockFunction]}
    id="passwordConfirm"
    label="Confirm Your Password"
    meta={
      Object {
        "error": null,
        "touched": false,
      }
    }
    name="passwordConfirm"
    placeholder=""
    required={true}
    type="password"
    value=""
  />
</Fragment>
`;
